AnalyticsKit.swift

```diff
@@ -1,36 +1,36 @@
 import Foundation
 
-@objc protocol AnalyticsKitProvider {
-    func applicationWillEnterForeground()
-    func applicationDidEnterBackground()
-    func applicationWillTerminate()
-    func uncaughtException(_ exception: NSException)
-    func logScreen(_ screenName: String)
-    func logScreen(_ screenName: String, withProperties properties: [String: Any])
+@objc public protocol AnalyticsKitProvider {
+    @objc optional func applicationWillEnterForeground()
+    @objc optional func applicationDidEnterBackground()
+    @objc optional func applicationWillTerminate()
+    @objc optional func uncaughtException(_ exception: NSException)
+    @objc optional func logScreen(_ screenName: String)
+    @objc optional func logScreen(_ screenName: String, withProperties properties: [String: Any])
```

Comment:
Many providers were ignoring some of the requirements for the Provider so I decide it might be best make some of these optional requirements